Calculator/ViewController.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  ViewController.swift
+//  Calculator
+//
+//  Created by Oleksii Kolomiiets on 4/4/18.
+//  Copyright © 2018 Oleksii Kolomiiets. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    @IBOutlet weak var displayLabel: UILabel!
+    
+    private var isInTheMiddleOfTyping = false
+    
+    var displayValue: Double {
+        get {
+            if let displayText = displayLabel.text, let doubleValue = Double(displayText) {
+                return doubleValue
+            } else {
+                return 0
+            }
+        }
+        set {
+            displayLabel.text = String(newValue)
+        }
+    }
+    
+    @IBAction func touchTheDigitButton(_ sender: UIButton) {
+        let digit = sender.currentTitle!
+        if isInTheMiddleOfTyping, displayLabel.text != "0", let textCurrentlyInDisplay = displayLabel.text, let inputedValue = Double(textCurrentlyInDisplay + digit) {
+            if isValid(valueForInput: inputedValue) {
+                displayLabel.text = textCurrentlyInDisplay + digit
+            }
+        } else {
+            displayLabel.text = digit
+            isInTheMiddleOfTyping = true
+        }
+    }
+      
+    private var brain = CalculatorBrain()
```

Comment:
Do you need a variable for 'brain' of it would be enough to have a constant?

Reply:
Yes, I needed a variable for 'brain', because 'CalculatorBrain' was a struct and in 'ViewController' I've been changing variables in that struct. But there are no requirements for value type, that is why I will make  'CalculatorBrain' a class.